Tests/GRPCTests/MutualTLSTests.swift

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2021, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import EchoImplementation
+import EchoModel
+@testable import GRPC
+import GRPCSampleData
+import Logging
+import NIOConcurrencyHelpers
+import NIOCore
+import NIOPosix
+import NIOSSL
+import XCTest
+
+class MutualTLSTests: GRPCTestCase {
+  enum ExpectedOutcome {
+    case success
+    case serverError
+    case clientError
+  }
+
+  func performTestWith(
+    _ serverTLSConfiguration: GRPCTLSConfiguration?,
+    _ clientTLSConfiguration: GRPCTLSConfiguration?,
+    expect expectedOutcome: ExpectedOutcome
+  ) throws {
+    // Setup the server.
+    let serverEventLoopGroup = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+    var serverConfiguration = Server.Configuration.default(
```

Comment:
Can you add a logger in here too? `GRPCTestCase` provides `self.serverLogger` which will emit all trace logs on test failures.